command/rootchain/deploy/deploy.go

```diff
@@ -506,17 +530,18 @@ func deployContracts(outputter command.OutputFormatter, client *jsonrpc.Client,
 		{
 			name:     customSupernetManagerName,
 			artifact: contractsapi.CustomSupernetManager,
+			hasProxy: true,
 		},
 	}
 
 	allContracts = append(tokenContracts, allContracts...)
 
 	g, ctx := errgroup.WithContext(cmdCtx)
-	results := make([]*deployContractResult, len(allContracts))
-	commandResults := make([]command.CommandResult, len(allContracts))
+	results := make(map[string]*deployContractResult, len(allContracts))
```

Comment:
We changed it to map when we were deploying proxies in separate goroutine, so we used the contract name to get result entry to obtain the address of the deployed implementation contract. But then Goran suggested we could do this in the same goroutine, so we changed that, but we left out the map. In any case, allContracts now doesn't contain all contracts- since proxies don't have their instances in this slice. We can switch back to slice, determine how many of them need proxies and make the slice of that size?